### PR TITLE
Added Switch Application

### PR DESCRIPTION
**Pull request by edjoperez**
Adds functionality to toggle between windows applications

image

image

It doenst use its  trigger, so it makes albert has a more natural experience when switch between applications or new ones. Like rofi

### CODE REVIEW

**Comment by ManuelSchneid3r** [comment]
Did you notice that there is https://github.com/albertlauncher/python/pull/5/files

**Comment by edjoperez** [comment]
The only difference is that this one gives you a more fluent experience when you select the app you want to launch, because it adds the applications that are already running and the applications that you would launch

**Comment by Raensul** [comment]
if i fv switching from phpstorm to chrome it is no focused window. i must click to chrome's window and then press ctrl+T for example

**Comment by husimo** [comment]
Hello. It works great here under Opensuse Leap 15.0.
Just a bug : 
- `Firefox` triggers the "Firefox" window where as `firefox` doesn't work.
- `Konsole` triggers the "konsole" window, the same as `Konsole`.
So it seems the script cannot properly detect windows where their name contains a uppercase letter.
Thank you

**Comment by ManuelSchneid3r** [comment]
Conversation locked until [poll](https://github.com/albertlauncher/albert/issues/123#issuecomment-413868346) finished.

File: SwitchApp.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        wm_class = args[2].decode('utf-8')
+        if window_filter(wm_class):
+            _list.append({
+                'wid':args[0].decode('utf-8'),
+                'wm_class':wm_class,
+                'wm_name':args[4].decode('utf-8') if len(args) > 4 else ''
+            })
+    return _list
+
+def window_search(filter=''):
```
Comment (ManuelSchneid3r) [inline SwitchApp.py]: as @husimo said please make this case insensitive.

**Comment by ManuelSchneid3r** [comment]
There is further the question how the heck we can get the correct icon paths. E.g. spotify has the icon name _spotify-client_. However it has to be possible because the window manager show the icons correctly.

**Comment by ed-fruty** [comment]
Works for almost all my applications, but won't switch to my `PHPStorm` running instance, only suggest to open a new one instance.

**Comment by monkeydri** [comment]
Does not work for me, SwitchyApp 1.2 activated and `wmctrl` installed but no existing window appear in the results.

**Comment by divyanshu013** [comment]
Hey guys, how do I use this feature for switching apps?

**Comment by ManuelSchneid3r** [comment]
I am not sure anymore is this extension shipped in 14.22? Otherwise you have to download and install it manually.

**Comment by offzale** [comment]
@ManuelSchneid3r is this included in version 0.16.1? It is not working for me installing the rpm package in fedora 29

**Comment by kuamatzin** [comment]
I just enabled the extension but it does not work, I already install wmctrl, do I need some extra step?

**Comment by Mimameid** [comment]
> Hey guys, how do I use this feature for switching apps?
Bump

**Comment by jason--liu** [comment]
Hey guys, Im still wondering how to trigger this feature? thanks.

**Comment by ManuelSchneid3r** [comment]
> I just enabled the extension but it does not work, I already install wmctrl, do I need some extra step?

What does stdout tell?

**Comment by rosaLux161** [comment]
Any updates on this?

**Comment by justpeanuts** [comment]
Sorry if this is a dumb question, but is this feature currently there in albert? - switch to an application window if it is already open instead of launching another instance (without using the seperate menu item to switch to an open window using the "window-switcher.py" python extension)

Thank you

**Comment by zoli** [comment]
> Sorry if this is a dumb question, but is this feature currently there in albert? - switch to an application window if it is already open instead of launching another instance (without using the seperate menu item to switch to an open window using the "window-switcher.py" python extension)
> 
> Thank you

Yes in python plugins there is a window switcher that you should enable. Then for example if you have firefox open, When you search firefox in albert it brings another entry other that switches to running instance of firefox.

**Comment by tuttiamore** [comment]
The problem appears to be related to faulty destructuring of the window names into variables - the code assumes that window names always consists of two elements seperated by a `.`, when in fact window names can consists of more than two elements.

A quick hack for people interested in making this work:
Change the code in the python module on line 28 from: 
`win_instance, win_class = win.wm_class.replace(' ', '-').split('.')`
to 
`win_instance, win_class = win.wm_class.replace(' ', '-').rsplit('.',1)`

**Comment by ManuelSchneid3r** [comment]
Please continue any discussion here: https://github.com/orgs/albertlauncher/discussions/1299

**Comment by ManuelSchneid3r** [comment]
https://github.com/albertlauncher/albert/issues/123#issuecomment-413868346

This poll decided on this plugin. If anyone interested in porting it to api v2.x please send a PR.

**Pull request merged**